test/e2e/test_override_job_user.py

```diff
@@ -234,3 +237,209 @@ def test_env_var_user_override(
+            deadline_client=deadline_client,
+            logs_client=boto3.client(
+                "logs",
+                config=botocore.config.Config(retries={"max_attempts": 10, "mode": "adaptive"}),
+            ),
+            expected_pattern="I am: config-override",
+        )
+
+        assert job.task_run_status == TaskStatus.SUCCEEDED
+
+        cmd_result = class_worker.send_command(
+            'sed -i \'s/posix_job_user = "config-override:job-override-group"/# posix_job_user = "user:group"/g\' /etc/amazon/deadline/worker.toml'
+        )
+        assert (
+            cmd_result.exit_code == 0
+        ), f"Resetting the job user override via CLI failed: {cmd_result}"
+
+    def test_env_var_user_override(
+        self,
+        deadline_resources,
+        class_worker: EC2InstanceWorker,
+        deadline_client: DeadlineClient,
+    ) -> None:
+
+        class_worker.stop_worker_service()
+
+        @backoff.on_exception(
+            backoff.constant,
+            Exception,
+            max_time=45,
+            interval=5,
+        )
+        def check_worker_service_stopped() -> None:
+            worker_status_cmd_response = class_worker.send_command(
+                "systemctl is-active deadline-worker"
+            )
+
+            assert worker_status_cmd_response.exit_code != 0
+            assert worker_status_cmd_response.stdout != "active"
+
+        check_worker_service_stopped()
+
+        cmd_result = class_worker.send_command(
+            'echo "Environment=DEADLINE_WORKER_POSIX_JOB_USER=env-override:job-override-group" >> /etc/systemd/system/deadline-worker.service.d/config.conf',
+        )
+
+        assert (
+            cmd_result.exit_code == 0
+        ), f"Failed to set DEADLINE_WORKER_POSIX_JOB_USER: {cmd_result}"
+
+        class_worker.send_command("systemctl daemon-reload")
+        class_worker.start_worker_service()
+
+        job = self.submit_whoami_job(
+            "environment override",
+            deadline_client,
+            deadline_resources.farm,
+            deadline_resources.queue_a,
+        )
+
+        job.wait_until_complete(client=deadline_client, max_retries=20)
+
+        job.assert_single_task_log_contains(
+            deadline_client=deadline_client,
+            logs_client=boto3.client(
+                "logs",
+                config=botocore.config.Config(retries={"max_attempts": 10, "mode": "adaptive"}),
+            ),
+            expected_pattern="I am: env-override",
+        )
+
+        assert job.task_run_status == TaskStatus.SUCCEEDED
+
+        cmd_result = class_worker.send_command(
+            "sed -i '/Environment=DEADLINE_WORKER_POSIX_JOB_USER=env-override/d' /etc/systemd/system/deadline-worker.service.d/config.conf"
+        )
+        assert (
+            cmd_result.exit_code == 0
+        ), f"Resetting the job user override via CLI failed: {cmd_result}"
+        class_worker.send_command("sudo systemctl daemon-reload")
```

Comment:
We need to do this because this worker is reused in future tests right? Maybe this should be in a `finally` block in case something goes wrong in the test

Reply:
This worker would be used if any future tests are added to this class. I can make it into a finally block.